### DISCUSSION TITLE
Can I add Or inside filter along with many And operators?

### DISCUSSION THREAD

**Discussion by pmt205** [Snow-Shell/servicenow-powershell#214] [Q&A] [answered]
I want to add and Or inside my filter along with many And operator, is there any way that I can do that?
For example: `<First condition> and (<Second condition> or <Third condition>) and <Fourth condition>`

I know in the Get-help there is a '-group' but looks like it just can do the filter like `(<First> and <Second>) or <Third>`

**Comment by gdbarron** [comment, marked as answer]
I built your example in the UI and the query was just as you have it, without the parenthesis.  a and b or c and d.  I haven't confirmed this, but it seems 'or' takes precedence over 'and' which is why group is needed to override.

**Reply by pmt205** [reply]
So does that '-group' will group all the condition before it? like in your example, it groups the state and description.
Because I'm still stuck a little bit at how to use the '-group', could you help to explain further on it? Thanks.

**Reply by gdbarron** [reply]
Let's take the original example.  If you have `a and b or c and d` and you want to represent it as `a and (b or c) and d`, you would use it as is since 'or' takes precedence.  But, if you wanted to represent `(a and b) or (c and d)`, this is where you would use group, `a and b group c and d`.  Effectively, 'or' no longer takes precedence.

**Reply by pmt205** [reply]
Nice, understood, thank you very much! And also thank you for developing this wonderful module, which helps me a lot in my work.
